[Jarveis Lam] iP

jrvslam opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

jrvslam committed:
Implemented Skeletal structure for project

jrvslam committed:
Added ability to store text entered by the user and return list of text given.

jrvslam committed:
Added ability to mark tasks as done

jrvslam committed:
Add support to track 3 tasks: ToDos, Deadlines, and Events

jrvslam committed:
Edit bye message

jrvslam committed:
Edited inputs and expected outputs

jrvslam committed:
Added exception handlers

jrvslam committed:
Added remove feature that deletes task from list

jrvslam committed:
Used enums to simplify task states

jrvslam committed:
Save tasks to a file in ./data/duke.txt

jrvslam committed:
Change name of chatbot from Duke to Rawr

jrvslam committed:
Add load function to load data from hard drive.

jrvslam committed:
Add date and time formatting so Duke can understand Date and Time

jrvslam committed:
Merge branch 'branch-Level-7' into master

# Conflicts:
#	src/main/java/Duke.java

jrvslam committed:
Merge branch 'branch-Level-8' into master

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java

jrvslam committed:
Update to allow saved data to follow date and time format

jrvslam committed:
Update code to be follow more OOP structure

jrvslam committed:
Update structure to follow OOP

jrvslam committed:
Package files into each own java packages

jrvslam committed:
Add unit testing component

jrvslam committed:
Fix saving function so that data will only be overwritten at end

jrvslam committed:
Add ability to find and search for items

jrvslam committed:
Add javadoc comments to different methods and classes

jrvslam committed:
Update code to fit code style

jrvslam committed:
Update javadoc for majority public classes and methods

jrvslam committed:
Remove saved duke logo

jrvslam committed:
Merge branch 'branch-A-JavaDoc' into master

jrvslam committed:
Merge branch 'branch-A-CodingStandard' into master

# Conflicts:
#	src/main/java/duke/command/ParseCommands.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/ui/Ui.java

jrvslam committed:
Merge branch 'branch-Level-9' into master

# Conflicts:
#	src/main/java/duke/command/ParseCommands.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/ui/Ui.java

dannn-tran commented on src/main/java/duke/command/ParseCommands.java:
```diff
@@ -0,0 +1,204 @@
+        }
+    }
+
+    private static CommandList changeToCommand (String command) throws DukeException {
```
Should you remove the space after the method name?
Additionally, should you use a more meaningful method name? For example `createCommandFromString`

dannn-tran commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,170 @@
+        return file;
+    }
+
+    private void populateList (ArrayList<Task> arr) {
```
Should you remove the space after method's name here as well?

dannn-tran commented on src/main/java/duke/task/Task_State.java:
```diff
@@ -0,0 +1,15 @@
+package duke.task;
+
+public enum Task_State {
```
Should you use Camel case to name this class name, for consistency?

dannn-tran commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,170 @@
+    /**
+     * This closes the FileWriter used by the chatbot in Storage.
+     */
+    public void closeFile () {
```
Should you remove the space after method's name?

dannn-tran commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,22 @@
+    private final String exception;
+
+    /**
+     * This is the constructor for a DukeException that takes in a String message.
+     * @param exception This is the String error message.
```
Should you have an empty line between the description and params?

dannn-tran commented on src/main/java/duke/command/ParseCommands.java:
```diff
@@ -0,0 +1,204 @@
+ */
+import java.util.ArrayList;
+
+public class ParseCommands {
```
Should you name the class as a noun instead? e.g. `CommandParser`

dannn-tran commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+    }
+
+    @Override
+    public String taskSave() {
```
Should you name this as `saveTask` instead?

dannn-tran reviewed (commented):
Generally very clean; well done!
Just some minor cosmetic changes :).

jrvslam committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

jrvslam committed:
Update build.gradle

jrvslam committed:
Clean up code to follow more Coding Standards

jrvslam committed:
Add GUI features

jrvslam committed:
Merge branch 'branch-Level-10' into master

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/command/CommandParser.java

Andrewzhang217 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,30 @@
+import java.util.Scanner;
```
Well done, naming is good and coding standard is strictly followed!

Andrewzhang217 commented on src/main/java/Task_State.java:
```diff
@@ -1,3 +1,13 @@
 public enum Task_State {
```
Should you use Pascalcase to name the enum? ie TaskState

jrvslam committed:
Add assertion statements and update gradle path

jrvslam committed:
Remove some unused imports

jrvslam committed:
Remove some unused imports

jrvslam committed:
Remove some unused imports

jrvslam committed:
Add comment

jrvslam committed:
Add assert to Storage when populating list from file.

jrvslam committed:
Improve code quality by abstracting out functions in CommandParser

jrvslam committed:
Merge pull request #3 from jrvslam/branch-A-Assertions

Branch a assertions

jrvslam committed:
Merge branch 'master' into branch-A-CodeQuality

jrvslam committed:
Merge pull request #4 from jrvslam/branch-A-CodeQuality

Branch a code quality

jrvslam committed:
Fix minor bug after merging branches

jrvslam committed:
Remove unused code

jrvslam committed:
Add reminders to show immediately at start and when command is
called. Add functionality to BYE command where application will
close when BYE is given

jrvslam committed:
Clean up Ui Class

jrvslam committed:
Merge branch 'branch-BCD-Extensions' into master

jrvslam committed:
Update GUI to fit more chatbot like structure

jrvslam committed:
Update GUI font and change profile picture

jrvslam committed:
Update GUI background and chat bubbles

jrvslam committed:
Update GUI

jrvslam committed:
Improve GUI

jrvslam committed:
Add help functionality

jrvslam committed:
Added Screenshots of GUI

jrvslam committed:
Add file via opload

jrvslam committed:
Update README.md

jrvslam committed:
Update README.md

jrvslam committed:
Update README.md

jrvslam committed:
Update README.md

jrvslam committed:
Add images for readme

jrvslam committed:
Update README.md

jrvslam committed:
Update README.md

jrvslam committed:
Update README.md

jrvslam committed:
Add logo

jrvslam committed:
Update _config.yml

jrvslam committed:
Update README.md

jrvslam committed:
Update README.md

jrvslam committed:
Update README.md

jrvslam committed:
Change Help.pmg

jrvslam committed:
Update help command to include the find command

jrvslam committed:
Update help.png

jrvslam committed:
Update README.md

jrvslam committed:
Update _config.yml

jrvslam committed:
Update README.md

jrvslam committed:
Add Ui.png for submission

jrvslam committed:
Update _config.yml

jrvslam committed:
Update README.md

jrvslam committed:
Create final jar for release

jrvslam committed:
Update README.md

jrvslam committed:
Update README.md

jrvslam committed:
Update README.md
